KeepingYouAwake/KYAAppController/KYAAppController.m

```diff
@@ -176,6 +188,20 @@ - (void)configureUserNotificationCenter
 
 #pragma mark - Device Power Monitoring
 
+- (void)checkPowerSource
+{
+    CFTypeRef powerSourceInfo = IOPSCopyPowerSourcesInfo();
+    CFArrayRef powerSources = IOPSCopyPowerSourcesList(powerSourceInfo);
```

Comment:
Implementation-wise this is certainly possible to solve it like that.

I would prefer to not add IOPowerSources code directly here in `KYAAppController`. This would probably be a better fit for `KYABatteryMonitor`. There is some pre-existing IOPowerSources code already, that could be extended, e.g. by adding concrete notifications for this connect/disconnect scenario.